tests/integration/test_solver_slippage.py

```diff
@@ -78,11 +78,14 @@ def setUp(self) -> None:
     def get_cte_rows(
         self,
         cte_name: str,
+        period: Optional[AccountingPeriod] = None,
         tx_hash: Optional[str] = None,
         v1_cache: Optional[str] = None,
         v2_cache: Optional[str] = None,
     ):
-        parameters = self.period.as_query_params()
+        parameters = (
+            self.period.as_query_params() if not period else period.as_query_params()
```

Comment:
This feels like a bit of an anti-patern where we are using two different periods for the same test.

Don't have enough context to know if this is a valid expectation to have, but this struct me as very weird when reading the code.

Reply:
I could have introduced a separate test file, but then this nice helper method would have to be migrated. The thing is that the class has a "global" accounting period for which we have been testing all the other queries. For individual transaction hashes, it might actually make sense to move them into their own test suite (although we would have to make bigger changes. I kinda like that the helper method is "robust" enough to take either the global configuration or an optional alternative (there are many optional parameters in this query). It just so happens that we haven't needed to alter them until now.

Reply:
Actually, it could be migrated out into a "common" file, the period would have to be supplied by every test that uses it. I wouldn't mind making this change here if you insist (but it will make the diff bigger).

Reply:
Honestly, I don't think this change is appropriate to make here because it will touch too much other code. I can prepare a follow up for this.

Reply:
I have move this. let me know if its better now.